File: Mixter/Domain/MessagePublished.cs
namespace Mixter.Domain
{
    public struct MessagePublished : IDomainEvent
    {
        public MessageId Id { get; private set; }

        public string Message { get; private set; }

        public MessagePublished(MessageId id, string message)
            : this()
        {
            Message = message;
            Id = id;
        }
    }
}

File: src/main/java/mixter/domain/message/MessagePublished.java
package mixter.domain.message;

import mixter.Event;
import mixter.UserId;

public class MessagePublished implements Event {
    private final MessageId messageId;
    private final String message;
    private final UserId authorId;

    public MessagePublished(MessageId messageId, String message, UserId authorId) {
        this.messageId = messageId;
        this.message = message;
        this.authorId = authorId;
    }

    public MessageId getMessageId() {
        return messageId;
    }

    public String getMessage() {
        return message;
    }

    public UserId getAuthorId() {
        return authorId;
    }
}
